p2p/host.go

```diff
@@ -384,6 +400,56 @@ func (host *HostV2) ConnectHostPeer(peer Peer) error {
 	return nil
 }
 
+// called when network starts listening on an addr
+func (host *HostV2) Listen(net libp2p_network.Network, addr ma.Multiaddr) {
+
+}
+
+// called when network stops listening on an addr
+func (host *HostV2) ListenClose(net libp2p_network.Network, addr ma.Multiaddr) {
+
+}
+
+// called when a connection opened
+func (host *HostV2) Connected(net libp2p_network.Network, conn libp2p_network.Conn) {
+	host.logger.Info().Interface("node", conn.RemotePeer()).Msg("peer connected")
+
+	for _, function := range host.onConnections {
+		if err := function(net, conn); err != nil {
+			host.logger.Error().Err(err).Interface("node", conn.RemotePeer()).Msg("failed on peer connected callback")
+		}
+	}
+}
+
+// called when a connection closed
+func (host *HostV2) Disconnected(net libp2p_network.Network, conn libp2p_network.Conn) {
+	host.logger.Info().Interface("node", conn.RemotePeer()).Msg("peer disconnected")
+
+	for _, function := range host.onDisconnects {
+		if err := function(conn); err != nil {
+			host.logger.Error().Err(err).Interface("node", conn.RemotePeer()).Msg("failed on peer disconnected callback")
+		}
+	}
+}
+
+// called when a stream opened
+func (host *HostV2) OpenedStream(net libp2p_network.Network, stream libp2p_network.Stream) {
```

Comment:
Do you think it will be a better name as `onOpenedStream`?

Reply:
Implement Notifiee interface